doc/developer-guides/virtio-console.rst

```diff
@@ -0,0 +1,183 @@
+.. virtio-console:
+
+Virtio-Console High-Level design
+################################
+
+The Virtio-console is a simple device for data input and output.  The
+console's virtio device ID is ``3`` and can have from 1 to 16 ports.
+Each port has a pair of input and output virtqueues used to communicate
+information between the Front End (FE) and Back end (BE) drivers.
+Currently the size of each virtqueue is 64 (configurable in the source
+code).  The FE driver will place empty buffers for incoming data onto
+the receiving virtqueue, and enqueue outgoing characters onto the
+transmitting virtqueue.
+
+A Virtio-console device has a pair of control IO virtqueues as well. The
+control virtqueues are used to communicate information between the
+device and the driver, including: ports being opened and closed on
+either side of the connection, indication from the host about whether a
+particular port is a console port, adding new ports, port
+hot-plug/unplug, indication from the guest about whether a port or a
+device was successfully added, or a port opened or closed.
+
+The virtio-console architecture diagram in ACRN is shown below.
+
+.. figure:: images/virtio-console-arch.png
+   :align: center
+   :width: 700px
+   :name: virtio-console-arch
+
+   Virtio-console architecture diagram
+
+
+Virtio-console is implemented as a virtio legacy device in the ACRN device
+model (DM), and is registered as a PCI virtio device to the guest OS. No changes
+are required in the frontend Linux virtio-console except that the guest kernel
+should be built with ``CONFIG_VIRTIO_CONSOLE=y``.
```

Comment:
@chejianj  there is no configuration option with the name ``CONFIG_VIRTIO_CONSOLE``.  What should this reference be?

Reply:
Ah, thanks.  I was expecting this to be a reference found in our hypervisor configuration options document (https://projectacrn.github.io/latest/reference/kconfig/index.html).